src/event_handler.cpp

```diff
@@ -182,15 +164,17 @@ void	EventHandler::HandlePollHupEvent(pollfd entry) {
 	}
 }
 
-void	EventHandler::Detach(pollfd event) {
-	std::cout << "connection hang up " << event.fd << std::endl;
+void	EventHandler::Detach(pollfd entry) {
+	std::cout << "connection hang up " << entry.fd << std::endl;
 	int target_index = 0;
 	for (int i = 0; i < (int)poll_fd_.size(); i++)
 	{
-		if (poll_fd_[i].fd == event.fd)
+		if (poll_fd_[i].fd == entry.fd)
 			target_index = i;
 	}
+	int target_fd = (poll_fd_.begin() + target_index)->fd;
 	poll_fd_.erase(poll_fd_.begin() + target_index);
```

Comment:
その2
こちらではもう一点、指摘。
target_fdの取得の仕方ですが、poll_fd_はvectorなので、これで良いです。
何ならこの処理、for文の中に書くともっと短縮できるかも。
```suggestion
		if (poll_fd_[index].fd == entry.fd)
			break ;
	}
	if (index == poll_fd_.size())
		return ;
	int target_fd = poll_fd_[index].fd;
	poll_fd_.erase(poll_fd_.begin() + index);
```